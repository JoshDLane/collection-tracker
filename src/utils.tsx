import * as eva from "@eva-design/eva";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";

export function UIKittenProvider(props: any) {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={eva.light}>
        {props.children}
      </ApplicationProvider>
    </>
  );
}
