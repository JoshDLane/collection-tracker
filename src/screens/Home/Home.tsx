import React from "react";
import { SafeAreaView, View } from "react-native";
import { Button, Divider, Layout, TopNavigation } from "@ui-kitten/components";
import { React$Node } from "../../utils";

// interface HomesScreenProps {
//   navigation: any;
// }

export default function HomeScreen(): React$Node {
  // const navigateDetails = () => {
  //   navigation.navigate("Details");
  // };
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <View>
        <Button onPress={() => {}}>EXAMPLE</Button>
      </View>
      {/* <TopNavigation title='MyApp' alignment='center'/>
      <Divider/>
      <Layout style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <Button onPress={navigateDetails}>OPEN DETAILS</Button>
      </Layout> */}
    </SafeAreaView>
  );
}
