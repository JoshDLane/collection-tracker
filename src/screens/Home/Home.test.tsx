import React from "react";
import renderer from "react-test-renderer";
import { UIKittenProvider } from "../../utils";
import Home from "./Home";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <UIKittenProvider>
        <Home />
      </UIKittenProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
