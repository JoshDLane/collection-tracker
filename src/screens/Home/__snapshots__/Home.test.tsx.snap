// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3366FF",
            "borderColor": "#3366FF",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
              },
            ]
          }
        >
          EXAMPLE
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
